server/plugin/command_test.go

```diff
@@ -333,3 +334,324 @@ func TestExecuteCommand(t *testing.T) {
 		})
 	}
 }
+
+func TestGetMutedUsernames(t *testing.T) {
+	mockKvStore, mockAPI, _, _, _ := GetTestSetup(t)
+	p := getPluginTest(mockAPI, mockKvStore)
+	userInfo, err := GetMockGHUserInfo(p)
+	assert.NoError(t, err)
+
+	tests := []struct {
+		name       string
+		setup      func()
+		assertions func(t *testing.T, result []string)
+	}{
+		{
+			name: "Error retrieving muted usernames",
+			setup: func() {
+				mockKvStore.EXPECT().Get("mockUserID-muted-users", gomock.Any()).Return(errors.New("error retrieving muted users")).Times(1)
+			},
+			assertions: func(t *testing.T, result []string) {
+				assert.Nil(t, result)
+			},
+		},
+		{
+			name: "No muted usernames set for user",
+			setup: func() {
+				mockKvStore.EXPECT().Get("mockUserID-muted-users", gomock.Any()).DoAndReturn(func(key string, value *[]byte) error {
+					*value = []byte("")
+					return nil
+				}).Times(1)
+			},
+			assertions: func(t *testing.T, result []string) {
+				assert.Equal(t, []string(nil), result)
+			},
+		},
+		{
+			name: "Successfully retrieves muted usernames",
+			setup: func() {
+				mutedUsernames := []byte("user1,user2,user3")
+				mockKvStore.EXPECT().Get("mockUserID-muted-users", gomock.Any()).DoAndReturn(func(key string, value *[]byte) error {
+					*value = mutedUsernames
+					return nil
+				}).Times(1)
+			},
+			assertions: func(t *testing.T, result []string) {
+				assert.Equal(t, []string{"user1", "user2", "user3"}, result)
+			},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.setup()
+
+			mutedUsernames := p.getMutedUsernames(userInfo)
+
+			tc.assertions(t, mutedUsernames)
+		})
+	}
+}
+
+func TestHandleMuteList(t *testing.T) {
+	mockKvStore, mockAPI, _, _, _ := GetTestSetup(t)
+	p := getPluginTest(mockAPI, mockKvStore)
+	userInfo, err := GetMockGHUserInfo(p)
+	assert.NoError(t, err)
+
+	tests := []struct {
+		name       string
+		setup      func()
+		assertions func(t *testing.T, result string)
+	}{
+		{
+			name: "Error retrieving muted usernames",
+			setup: func() {
+				mockKvStore.EXPECT().Get("mockUserID-muted-users", gomock.Any()).Return(errors.New("error retrieving muted users")).Times(1)
+			},
+			assertions: func(t *testing.T, result string) {
+				assert.Equal(t, "You have no muted users", result)
```

Comment:
Why would `p.handleMuteList` return "You have no muted users" on a failure?

Reply:
@wiggin77 I have updated the getMutedUser function to return a proper response. Please have a look